### DISCUSSION TITLE
Next link  making a full page refresh?

### DISCUSSION THREAD

**Discussion by vkotu** [vercel/next.js#45537] [Help] [answered]
Hi Team - Not sure what Im missing but Next link [here](https://github.com/vkotu/nextjs-intro/blob/main/pages/notes/index.jsx#L28) is most of the time does a full refresh and causes issues because of that while accessing the state set in _app. 

It doesn't happen with Link inside this [Nav](https://github.com/vkotu/nextjs-intro/blob/main/src/components/nav.jsx#L36)

Here is my code [base](https://github.com/vkotu/nextjs-intro), a pretty simple app with some page navigations.

Can someone please help what is wrong with my code?

![Link Issue](https://user-images.githubusercontent.com/8981254/216466205-531249e8-5290-4fcf-a854-5d34943c591b.gif)

**Comment by icyJoseph** [comment, marked as answer]
Hi,

There's quite a bit that's not correct, or that I wouldn't call best practice to say the least.

Your app is failing because when the notes page is compiled, the notes list is:

```
[
  { id: 1675410902850, title: 'Note 0' },
  { id: 1675410902851, title: 'Note 1' },
  { id: 1675410902852, title: 'Note 2' },
  { id: 1675410902853, title: 'Note 3' },
  { id: 1675410902854, title: 'Note 4' },
  { id: 1675410902855, title: 'Note 5' },
  { id: 1675410902856, title: 'Note 6' },
  { id: 1675410902857, title: 'Note 7' },
  { id: 1675410902858, title: 'Note 8' },
  { id: 1675410902859, title: 'Note 9' },
  { id: 1675410902860, title: 'Note 10' },
  { id: 1675410902861, title: 'Note 11' },
  { id: 1675410902862, title: 'Note 12' },
  { id: 1675410902863, title: 'Note 13' },
  { id: 1675410902864, title: 'Note 14' }
]
```

But when your API route is compiled, the notes list is:

```
[
  { id: 1675410909254, title: 'Note 0' },
  { id: 1675410909255, title: 'Note 1' },
  { id: 1675410909256, title: 'Note 2' },
  { id: 1675410909257, title: 'Note 3' },
  { id: 1675410909258, title: 'Note 4' },
  { id: 1675410909259, title: 'Note 5' },
  { id: 1675410909260, title: 'Note 6' },
  { id: 1675410909261, title: 'Note 7' },
  { id: 1675410909262, title: 'Note 8' },
  { id: 1675410909263, title: 'Note 9' },
  { id: 1675410909264, title: 'Note 10' },
  { id: 1675410909265, title: 'Note 11' },
  { id: 1675410909266, title: 'Note 12' },
  { id: 1675410909267, title: 'Note 13' },
  { id: 1675410909268, title: 'Note 14' }
]
```

When you click on Note 0, with id 1675410902850, it is nowhere to be found, so your app crashes out because of bad management on `getServerSideProps`.

Your key assumption is that notes is stable through out the APP, but that's not correct, it is compiled with each file that needs it, and that yields a different list of notes, since you make them dependant on Date.

## Additional tips

You are also reaching out for API routes from within GSSP and that's not a very good idea. Read the docs on this matter: https://nextjs.org/docs/basic-features/data-fetching/get-server-side-props#getserversideprops-or-api-routes

You are manipulating the response object in GSSP, rather, you should use `redirect` and `notFound`. 

- https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props#notfound
- https://nextjs.org/docs/api-reference/data-fetching/get-server-side-props#redirect

**Reply by vkotu** [reply]
Thanks a ton, @icyJoseph . I understood whats going and learned better.

**Reply by vkotu** [reply]
@icyJoseph As you know the structure, Can you also please suggest how can I avoid GSSP props call on the subsequent page navigations between home and /notes? Now I see GSSP called every time I go back and forth.
